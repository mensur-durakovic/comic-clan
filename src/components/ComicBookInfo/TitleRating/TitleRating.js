import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import StarRating from "../../Rating/Rating";

const TitleAndRatingWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 10px;

  @media (min-width: 320px) and (max-width: 480px) {
    display: block;
  }
`;

const BookTitle = styled.div`
  font-weight: 500;
  font-size: 32px;
  line-height: 37px;
  color: #aaaaaa;
`;

const TitleRating = React.memo((props) => {
  const { name, year, rating } = props;
  return (
    <TitleAndRatingWrapper>
      <BookTitle>
        {name}{` (${year})`}
      </BookTitle>
      <StarRating rating={rating} />
    </TitleAndRatingWrapper>
  );
});

TitleRating.propTypes = {
  name: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired
};

export default TitleRating;
