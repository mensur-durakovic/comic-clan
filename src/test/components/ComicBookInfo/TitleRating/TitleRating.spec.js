import React from "react";
import renderer from "react-test-renderer";
import TitleRating from "../../../../components/ComicBookInfo/TitleRating/TitleRating";

test("TitleRating component", () => {
  const tree = renderer
    .create(<TitleRating name="test" year={2020} rating={5} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
