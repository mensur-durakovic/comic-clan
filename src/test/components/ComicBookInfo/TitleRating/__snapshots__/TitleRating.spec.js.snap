// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleRating component 1`] = `
<div
  className="sc-AxhUy eCJffQ"
>
  <div
    className="sc-AxgMl fCrwUY"
  >
    test
     (2020)
  </div>
  <div
    className="sc-AxjAm fCkbWQ"
  >
    <div
      className="sc-AxirZ sc-AxiKw iChZue"
    />
    <div
      className="sc-AxirZ sc-AxiKw iChZue"
    />
    <div
      className="sc-AxirZ sc-AxiKw iChZue"
    />
    <div
      className="sc-AxirZ sc-AxiKw iChZue"
    />
    <div
      className="sc-AxirZ sc-AxiKw iChZue"
    />
  </div>
</div>
`;
